containers/Loans/Loans.tsx

```diff
@@ -195,36 +129,34 @@ function Container() {
 					});
 				};
 
-				const onLoanCreated = async (owner: string, id: string) => {
-					const loan = await fetchLoan(owner, id);
+				const onLoanCreated = async (_address: string, id: BigNumber) => {
+					const loan = await getLoan({
+						id: id.toNumber(),
+						loanContract,
+						loanStateContract,
+						isL2,
+						address,
+					});
 					setLoans((loans) => [loan, ...loans]);
 				};
 
-				const onLoanClosed = (owner: string, id: string) => {
+				const onLoanClosed = (_owner: string, id: BigNumber) => {
 					setLoans((loans) => loans.filter((loan) => !loan.id.eq(id)));
 				};
 
-				const onCollateralDeposited = async (
-					owner: string,
-					id: string,
-					amount: ethers.BigNumber
-				) => {
+				const onCollateralDeposited = async (owner: string, id: BigNumber, amount: BigNumber) => {
 					setLoans((loans) =>
 						loans.map((loan) => {
 							if (loan.id.eq(id)) {
-								loan.collateral = loan.collateral.add(amount);
+								return { ...loan, collateral: loan.collateral.add(amount) };
 							}
 							return loan;
 						})
 					);
 					await updateLoan(owner, id);
 				};
 
-				const onCollateralWithdrawn = async (
-					owner: string,
-					id: string,
-					amount: ethers.BigNumber
-				) => {
+				const onCollateralWithdrawn = async (owner: string, id: BigNumber, amount: BigNumber) => {
```

Comment:
why async func?

Reply:
All of the event handlers that updates the loan are async functions. You can see it `await updateLoan(owner, id);` a few lines below.. It is a fair point that it isn't really necessary since we don't care about the result of calling updateLoan. But it also doesn't hurt and at least communicates that the function is doing something asynchronous.

Happy to remove async + await for those functions if you feel strongly about it, have other reasoning I'm not thinking about

Reply:
no was just wondering. all good thanks :)